add docs for Assistant Memories (#1658)
semgrep/semgrep-docs: docs/semgrep-assistant/getting-started.md

  ![ Policies modes](/img/semgrep-assistant-comment.png)
* You have selected PR/MR comments in **Semgrep AppSec Platform > Settings > Deployment** in the **Code** section.

### Add Memories (beta)

Assistant Memories allows users to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.

> > Assistant Memories is currently in **public beta**. Please contact Semgrep to obtain access to this feature.

Reviewer: You do not need to contact us to obtain access to the feature.

  ![ Policies modes](/img/semgrep-assistant-comment.png)
* You have selected PR/MR comments in **Semgrep AppSec Platform > Settings > Deployment** in the **Code** section.

### Add Memories (beta)

> Assistant Memories allows users to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.

Reviewer 2: ```suggestion
Assistant Memories allow users to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.
```
Not sure if single or plural here

Reviewer 2: Is it possible to be more specific than “users” about who can make these changes? Might it be Semgrep admins or more generally, AppSec teams? When I read this I wonder if a developer can add custom instructions.

Author: Basically, anyone who can see the suggested fix can create a memory, so it's not just admin. I'll change this to "AppSec teams and developers"

> Assistant Memories allow AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix, you can provide feedback by adding custom instructions.

Author: ```suggestion
Assistant Memories allows AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix, you can provide feedback by adding custom instructions.
```

> Assistant Memories allow AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.

Reviewer 3: is it `allow` or `allows` considering the feature is a single entity (i.e. "GitHub Teams is a tier..." and not "GitHub Teams are a tier")?

> Assistant Memories allow AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.

Reviewer 3: ```suggestion
Assistant Memories allow AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix, you can provide feedback by adding custom instructions.
```

> Assistant Memories allow AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.

Author: ```suggestion
Assistant Memories allows AppSec teams and developers to tailor Assistant's remediation guidance to their organization's standards and defaults on a per-project, per-rule basis. Whenever Assistant gives a suggested fix,  users can provide feedback by adding custom instructions.
```

Memories are enabled by default for all Assistant users.

To add a Memory:

> 1. Identify the specific instance of **Assistant's suggested fix** that you want to modify. Click **Customize fix** to open an input box, and enter your preferred remediation approaches and secure defaults for the project.

Reviewer 2: How prescriptive does the submitted, customized fix need to be? I wonder if the user can enter more general instructions like, “Fix it by using X” rather than explicitly writing the code that should replaced in line. I think it’s the former based on what I’ve seen and if that’s true, I think it would be valuable to explain that here.

Author: Yes, people can enter fairly general instructions, so I'll add an example of that

> 1. Identify the specific instance of **Assistant's suggested fix** that you want to modify. Click **Customize fix** to open an input box, and enter your preferred remediation approaches and secure defaults for the project. The suggestion you provide can be as general as "Use AWS Secrets Manager to manage secrets."

Reviewer 3: ```suggestion
1. Identify the specific instance of **Assistant's suggested fix** that you want to modify. Click **Customize fix** to open an input box, and enter your preferred remediation approaches and secure defaults for the project. The suggestion you provide can be as general as "Use AWS Secrets Manager to manage secrets."
```

I believe this has to be done in the finding's detail page? or is it the rule's page?

>    ![Assistant’s suggested fix for a hard-coded secret in the user’s code](/img/memories-1.png#md-width)

Reviewer 3: ```suggestion
   ![Assistant’s suggested fix for a hardcoded secret in the user’s code](/img/memories-1.png#md-width)
```
I am sorry to report that Google just wants to be different from MS and insists on hardcoded, not hard-coded. https://developers.google.com/style/word-list#hardcode

>    ***Figure***. Assistant’s suggested fix for a hard-coded secret in the user’s code.

Reviewer 3: ```suggestion
   ***Figure***. Assistant’s suggested fix for a hardcoded secret in the user’s code.
```
